languages/python/sqlalchemy-oso/sqlalchemy_oso/session.py

```diff
@@ -45,44 +46,47 @@ def set_get_session(oso: Oso, get_session_func):
     oso.register_constant(_OsoSession, "OsoSession")
 
 
-@event.listens_for(Query, "before_compile", retval=True)
-def _before_compile(query):
-    """Enable before compile hook."""
-    return _authorize_query(query)
+if not AT_LEAST_SQLALCHEMY_VERSION_1_4:
```

Comment:
Not sure if there's a better way to conditionally set up the event hooks based on the SQLAlchemy version, but this seems to work.

Reply:
Changed to `try...except` in efef45c